src/shogun/base/SGObject.cpp

```diff
@@ -1092,26 +1092,23 @@ void CSGObject::init_auto_params()
 	}
 }
 
+CSGObject* CSGObject::get(const std::string& name, index_t index) const
+{
+	return sgo_details::get_by_tag(this, name, std::move(sgo_details::GetByNameIndex(index)));
+}
+
 CSGObject* CSGObject::get(const std::string& name, std::nothrow_t) const
     noexcept
 {
-	if (auto* result = get_sgobject_type_dispatcher<CDistance>(name))
-		return result;
-	if (auto* result = get_sgobject_type_dispatcher<CKernel>(name))
-		return result;
-	if (auto* result = get_sgobject_type_dispatcher<CFeatures>(name))
-		return result;
-	if (auto* result = get_sgobject_type_dispatcher<CLabels>(name))
-		return result;
-	return nullptr;
+	return sgo_details::get_by_tag(this, name, std::move(sgo_details::GetByName()));
 }
 
 CSGObject* CSGObject::get(const std::string& name) const noexcept(false)
 {
 	auto* result = get(name, std::nothrow);
 	if (result == nullptr)
 	{
-		if (self->map.find(BaseTag(name)) != self->map.end())
+		if (has(name))
```

Comment:
@gf712 btw I changed this as I think it is cleaner to read. Maybe we want to update the implementation of that at some point ?

Reply:
I am not sure which implementation you mean?

Reply:
the one of `has`